content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -1,8 +1,148 @@
+  default
+
+A `.dvc` file dependency entry consists of a these possible fields:
+
+- `path`: Path to the dependency, relative to the `wdir` path (always present)
+- `md5`: MD5 hash for the dependency (most [stages](/doc/command-reference/run))
+- `repo`: This entry is only for external dependencies created with
+  `dvc import`, and can contains the following fields:
+
+  - `url`: URL of Git repository with source DVC project
+  - `rev`: Only present when the `--rev` option of `dvc import` is used.
+    Specific commit hash, branch or tag name, etc. (a
+    [Git revision](https://git-scm.com/docs/revisions)) used to import the
+    dependency from.
+  - `rev_lock`: Git commit hash of the external <abbr>DVC repository</abbr> at
+    the time of importing or updating (with `dvc update`) the dependency.
+
+  > See the examples in
+  > [External Dependencies](/doc/user-guide/external-dependencies) for more
+  > info.
+
+Note that comments can be added to `.dvc` files and `dvc.yaml` using the
+`# comment` syntax.
+
+> `.dvc` file comments are preserved among executions of the `dvc repro` and
+> `dvc commit` commands, but not when a `.dvc` file is overwritten by
+> `dvc add`,`dvc import`, or `dvc import-url`.
+
+## dvc.yaml files
+
+When you add commands to a pipeline with `dvc run`, the `dvc.yaml` file is
+created or updated. Here's a simple example:
+
+```yaml
+stages:
+  stageone:
+    cmd: python cmd.py input.data output.data metrics.json
+    deps:
+    - cmd.py
+    - input.data
+    outs:
+    - output.data
+    metrics:
+    - metrics.json
+  stagetwo:
+    cmd: python ...
+    ...
+```
+
+`dvc.yaml` files consists of a group of `stages` with names provided explicitly
+by the user with the `--name` (`-n`) option of `dvc run`.
+
+Each stage can contain the following fields:
+
+- `cmd`: Executable command defined in this stage
+- `deps`: List of <abbr>dependencies</abbr> for this stage
+- `outs`: List of <abbr>outputs</abbr> for this stage
+- `params` (optional): List of the [parameter](/doc/command-reference/params)
+  names and their current values
+- `metric` (optional): List of [metric](/doc/command-reference/metrics) files
+- `frozen` (optional): Whether or not this stage is frozen from reproduction
+- `always_changed` (optional) : Whether or not this stage is considered as
+  changed by commands such as `dvc status` and `dvc repro`. `false` by default
+
+An output entry consists of these fields:
+
+- `md5`: Hash value for the output file
+- `path`: Path to the output in the <abbr>workspace</abbr>, relative to the
+  location of the `.dvc` file
+- `cache` (optional): Whether or not DVC should cache the output. `true` by
+  default
+
+A `dvc.yaml` dependency entry consists of a these possible fields:
+
+- `path`: Path to the dependency, relative to the `wdir` path (always present)
+- `md5`: MD5 hash for the dependency (most [stages](/doc/command-reference/run))
+- `etag`: Strong ETag response header (only HTTP <abbr>external
```

Comment:
Both optional.

Reply:
Not optional, `dvc.yaml` cannot contain `md5` or `etag` at all.

Reply:
Ugh good point. I need to make a new dvc.lock section. I'm going to extract this task...

Reply:
Extracted to #1422 for now, let's pretend dvc.yaml combines dvc.lock for now... Thanks